src/main/kotlin/calculator/StringAddCalculator.kt

```diff
@@ -0,0 +1,34 @@
+package calculator
+
+class StringAddCalculator {
+
+    private val splitterFactory = SplitterFactory()
+    fun add(text: String?): Int {
+        if (text.isNullOrBlank()) {
+            return DEFAULT_RESULT_VALUE
+        }
+        val splitter = splitterFactory.getSplitter(text)
+        return sum(splitter.split(text))
+    }
+
+    private fun sum(tokens: List<String>): Int = tokens.sumOf { toInt(it) }
+    private fun toInt(text: String): Int {
+        try {
+            val result = text.toInt()
+            validatePlus(result)
+            return result
+        } catch (e: NumberFormatException) {
+            throw RuntimeException("입력값은 숫자만 가능합니다.", e)
+        }
```

Comment:
```suggestion
        val result = text.toIntOrNull() ?: throw RuntimeException("입력값은 숫자만 가능합니다.")
        validatePlus(result)
        return result
```

`toIntOrNull()`을 사용하여 리팩토링이 가능해보여요!